src/hotspot/share/gc/g1/g1CollectionSet.cpp

```diff
@@ -326,10 +326,14 @@ void G1CollectionSet::finalize_old_part(double time_remaining_ms) {
     G1CollectionCandidateRegionList initial_old_regions;
     assert(_optional_old_regions.length() == 0, "must be");
 
-    time_remaining_ms = _policy->select_candidates_from_marking(&candidates()->marking_regions(),
-                                                                time_remaining_ms,
-                                                                &initial_old_regions,
-                                                                &_optional_old_regions);
+    if (!collector_state()->in_young_gc_before_mixed()) {
```

Comment:
Could this be affirmative, sth like `in_mixed_phase`?

Reply:
I was thinking this is more convenient with pinned regions later on. I will change this for now, and change again if necessary.